xonsh/history/base.py

```diff
@@ -163,3 +169,36 @@ def clear(self):
         memory.
         """
         pass
+
+    @functools.cached_property
+    def ignore_regex(self):
+        self.history_ignore_regex = None
```

Comment:
you don't have to store it in an attribute. that will be handled by the decorator